docs/write_test/guidelines.rst

```diff
@@ -233,6 +234,27 @@ Learn more from `examples
 <https://github.com/microsoft/lisa/tree/main/examples/testsuites>`__ and
 `assertpy document <https://github.com/assertpy/assertpy#readme>`__.
 
+Tips for writing code
+--------------------------
+
+Before Python 3.6, you had two main ways of embedding Python expressions inside string
+literals for formatting: %-formatting and str.format(). These ways have some
+limitations, so they are not recommended to use.
```

Comment:
```
Prior to Python 3.6, there were two methods for `string interpolation`_: %-formatting (sometimes referred to as
C-strings) and ``str.format()`` (along with the related ``format()`` and ``object.__format__()``. These methods
have some limitations, so their use should be limited when support for Python < 3.6 is not required.

.. _string interpolation:  https://en.wikipedia.org/wiki/String_interpolation
```

Reply:
The whole thing is not worth to mention, because we're using <3.8. The key reasons to use f-string, because its readability is better.